src/protocol/sort/multi_bit_permutation.rs

```diff
@@ -46,7 +49,7 @@ pub async fn multi_bit_permutation<
     let equality_checks = try_join_all(
         (0..num_records)
             .zip(repeat(ctx.set_total_records(num_records)))
-            .map(|(i, ctx)| check_everything(ctx, i, input)),
+            .map(|(i, ctx)| check_everything(ctx, i, input[i].as_slice())),
     )
```

Comment:
I'm surprised clippy didn't yell at you about this loop. When I write code like this, it usually prefers me to just loop over the thing I'm indexing into, rather than loop over an int, and just use that int to index into stuff.

Reply:
Changed it